astropy/coordinates/matching.py

```diff
@@ -300,10 +300,11 @@ def search_around_sky(coords1, coords2, seplimit, storekdtree='_kdtree_sky'):
         The on-sky separation between the coordinates. Shape matches ``idx1``
         and ``idx2``.
     dist3d : `~astropy.units.Quantity`
-        The 3D distance between the coordinates. Shape matches ``idx1`` and
-        ``idx2``. If either ``coords1`` or ``coords2`` don't have a distance,
-        this is the 3D distance on the unit sphere, rather than a physical
-        distance.
+        The 3D distance between the coordinates. Shape matches ``idx1``
+        and ``idx2``; the unit is that of ``coords1``.
+        If either ``coords1`` or ``coords2`` don't have a distance,
+        this is the 3D distance on the unit sphere, rather than a
+        physical distance.
```

Comment:
True nitpicking, but what if `coords1` has a distance and `coords2` has not, and the result is empty? In that case, should the unit be dimensionless?